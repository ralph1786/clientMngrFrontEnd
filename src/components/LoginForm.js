import React, { Component } from "react";
import "./LoginForm.scss";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { loginProvider } from "../actions/authActions";

class LoginForm extends Component {
  state = {
    username: "",
    password: ""
  };

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  handleSubmit = e => {
    e.preventDefault();
    this.props.loginProvider(this.state);
  };

  render() {
    // console.log(this.props.provider);
    return (
      <div className="login animated flipInY">
        <h1>Welcome Provider</h1>
        <form onSubmit={this.handleSubmit}>
          <label>Username</label>
          <input
            type="text"
            value={this.state.username}
            name="username"
            onChange={this.handleChange}
          />
          <br />
          <label>Password</label>
          <input
            type="password"
            value={this.state.password}
            name="password"
            onChange={this.handleChange}
          />
          <br />
          <button>Log In</button>
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    provider: state.provider
  };
};

const mapDispatchToProps = dispatch => {
  return {
    loginProvider: providerObj => dispatch(loginProvider(providerObj))
  };
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(LoginForm)
);
