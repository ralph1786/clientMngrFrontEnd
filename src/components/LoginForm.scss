.login {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-template-rows: repeat(12, 1fr);
}

.login h1 {
  grid-column: 5/9;
  text-align: center;
  grid-row: 3/4;
}

.login form {
  -webkit-box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  -moz-box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  background-color: rgb(32, 221, 158);
  grid-column: 5/9;
  grid-row: 4/8;
  text-align: center;
  font-size: 2em;
  margin: 10px;
}

.login form label,
.login form input {
  margin-top: 50px;
  padding: 10px;
  height: 18px;
}

.login form input {
  margin-left: 5px;
  width: 50%;
  font-size: 0.8em;
  outline: none;
}

.login form label {
  color: rgb(80, 78, 78);
}

.login form button {
  width: 50%;
  height: 10%;
  font-size: 20px;
  margin-top: 50px;
  border-radius: 25px;
  outline: none;
}

.login form button:hover {
  cursor: pointer;
  background-color: rgb(250, 127, 250);
  color: white;
  transition: all 0.4s ease-in;
}
