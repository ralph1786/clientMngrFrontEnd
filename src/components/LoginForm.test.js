import React from "react";
import { shallow } from "enzyme";
import { LoginForm } from "./LoginForm";
import sinon from "sinon";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<LoginForm />);
});

it("provides snapshot of LoginFrom", () => {
  expect(wrapper).toMatchSnapshot();
});

// it("checks the form is submitted", () => {
//   const spy = sinon.spy();
//   const fakeFunc = {
//       preventDefault: () => {

//       }
//   }
//   wrapper.setProps({loginProvider: () => {}})
//     wrapper.find("form").simulate("submit", fakeFunc)
//   expect(spy).toHaveBeenCalled();
// });
