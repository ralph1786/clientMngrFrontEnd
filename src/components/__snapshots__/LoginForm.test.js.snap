// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`provides snapshot of LoginFrom 1`] = `
<div
  className="login animated flipInY"
>
  <h1>
    <Message />
     Provider
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label>
      Username
    </label>
    <input
      name="username"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <label>
      Password
    </label>
    <input
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <br />
    <button>
      Log In
    </button>
  </form>
</div>
`;
